sklearn/cluster/tests/test_optics.py

```diff
@@ -95,27 +95,18 @@ def test_dbscan_optics_parity():
     op = OPTICS(min_samples=10).fit(X)
     core_optics, labels_optics = op.extract_dbscan(0.3)
 
-    op_h = metrics.homogeneity_score(labels_true, labels_optics)
-    op_c = metrics.completeness_score(labels_true, labels_optics)
-    op_V = metrics.v_measure_score(labels_true, labels_optics)
-    op_m = metrics.adjusted_mutual_info_score(labels_true, labels_optics)
-    op_s = metrics.silhouette_score(X, labels_optics)
-
-    op_metrics = np.array([op_h, op_c, op_V, op_m, op_s])
-
-    # calculate dbscan labels and cluster metrics
+    # calculate dbscan labels
     db = DBSCAN(eps=0.3, min_samples=10).fit(X)
 
-    db_h = metrics.homogeneity_score(labels_true, db.labels_)
-    db_c = metrics.completeness_score(labels_true, db.labels_)
-    db_V = metrics.v_measure_score(labels_true, db.labels_)
-    db_m = metrics.adjusted_mutual_info_score(labels_true, db.labels_)
-    db_s = metrics.silhouette_score(X, db.labels_)
+    confusion_m = contingency_matrix(db.labels_, labels_optics)
+    agree = np.max(confusion_m, axis=1)
+    disagree = np.sum(confusion_m) - np.sum(agree)
 
-    db_metrics = np.array([db_h, db_c, db_V, db_m, db_s])
+    # verify core_labels match
+    assert_array_equal(core_optics, db.core_sample_indices_)
 
-    # compare metrics
-    assert_allclose(db_metrics, op_metrics, atol=0.02)
+    # verify label mismatch is < 5 labels
+    assert_less(disagree, 5)
```

Comment:
Cool! Btw, these days, with `pytest`, we'd prefer `assert disagree < 5`. (I've not mentioned this given that this PR came into our world long before pytest did.)

Reply:
ok, thanks-- fixed.